src/components/database.rs

```diff
@@ -14,13 +14,41 @@ use crate::entities::{
 
 pub type DBConnection = Pool<Postgres>;
 
+#[cfg_attr(test, faux::create)]
 #[derive(Clone)]
 pub struct DBRepositories {
     pub friendships: FriendshipsRepository,
     pub friendship_history: FriendshipHistoryRepository,
     pub user_features: UserFeaturesRepository,
 }
 
+#[cfg_attr(test, faux::methods)]
+impl DBRepositories {
+    pub fn new(
+        friendships: FriendshipsRepository,
+        friendship_history: FriendshipHistoryRepository,
+        user_features: UserFeaturesRepository,
+    ) -> Self {
+        Self {
+            friendships,
+            friendship_history,
+            user_features,
+        }
+    }
+
+    pub fn get_friendships(&self) -> FriendshipsRepository {
+        self.friendships.clone()
```

Comment:
let's avoid cloning structs. We can return a reference to struct and do `&self.friendships`